Support extensions for graphql_sync

tbsf opened the pull request:
(This is a work in progress.)

These changes seemed to work with the following test:

```
from ariadne.types import ExtensionSync
...
class TimestampExtension(ExtensionSync):
    def format(self):
        return {"timestamp": datetime.now().isoformat()}
...
graphql_sync(
    schema,
    ...
    extensions=[TimestampExtension]
```

This also worked if I left off extensions as a kwarg in graphql_sync's call to execute so there may be other test cases that should be checked with this. Is this sufficient to support extensions for synchronous calls or is there another issue that needs to be handled in order for this to work correctly?

rafalp commented:
Looking good so far! I guess PR should also include the same tests harness that async extensions do. We will likely also need to update the docs for this, and also make a decision, what do we want with extensions we already provide as part of `ariadne.contrib` that are currently async only.

tbsf commented:
@rafalp I updated the tests I found using extensions with sync versions. I also updated the `ariadne.contrib` extensions with sync versions. Hopefully this is the direction you want to go in. 

I also added werkzeug for the tests I created and extended the client to match what is returned by starlette. You should be able to use this pattern for other tests instead of middleware, but I wasn't sure if these were equivalent.

tbsf commented:
@rafalp Is there anything else that needs to be done for this to be merged in for 0.0.7? I have added tests and documentation.

rafalp commented:
@tbsf Sorry about not coming back to you 😅We are now hyper-occupied with other things so work on 0.7 got put on backburner, but I will try to do a review of this sometime this week.

rafalp commented on ariadne/graphql.py:
```diff
@@ -114,59 +114,73 @@ def graphql_sync(
-            data.get("operationName"),
-        )
+    extension_manager = ExtensionManager(extensions)
+    with extension_manager.request(context_value):
```
In `graphql` there's an empty line between extension_manager and context: 

```suggestion

    with extension_manager.request(context_value):
```

rafalp commented on ariadne/types.py:
```diff
@@ -49,6 +49,12 @@ def format(self) -> dict:
 
+class ExtensionSync(Extension):
+    def resolve(self, next_: Resolver, parent: Any, info: GraphQLResolveInfo, **kwargs):
+        result = next_(parent, info, **kwargs)
```
```suggestion
        return next_(parent, info, **kwargs)
```

rafalp commented on ariadne/types.py:
```diff
@@ -49,6 +49,12 @@ def format(self) -> dict:
+class ExtensionSync(Extension):
+    def resolve(self, next_: Resolver, parent: Any, info: GraphQLResolveInfo, **kwargs):
+        result = next_(parent, info, **kwargs)
+        return result
```
```suggestion
```

rafalp commented on ariadne/wsgi.py:
```diff
@@ -173,6 +181,14 @@ def get_context_for_request(self, environ: dict) -> Optional[ContextValue]:
+        self, environ: dict, context: Optional[ContextValue]
+    ) -> ExtensionList:
+        if callable(self.extensions):
+            extensions = self.extensions(environ, context)
```
```suggestion
            return self.extensions(environ, context)
```

rafalp commented on ariadne/wsgi.py:
```diff
@@ -173,6 +181,14 @@ def get_context_for_request(self, environ: dict) -> Optional[ContextValue]:
+    ) -> ExtensionList:
+        if callable(self.extensions):
+            extensions = self.extensions(environ, context)
+            return extensions
```
```suggestion
```

tbsf commented:
@rafalp I've updated with your suggested changes. Any suggestions how to fix the code coverage issues?

rafalp commented:
There's not much that can be done here. Creating extension that does nothing on its own (but inherits `ExtensionSync`) and running query with it could bump the coverage somewhat, and give us harness for default `resolve` implementation for sync extensions.

Besides adding such test, I wouldn't obsess about codecov that much ;)

rafalp commented:
@tbsf Thanks! I'll also merge docs updates, but I still need to write docs for directives before releasing 0.7.
